path/path.go

```diff
@@ -55,19 +87,22 @@ func (s *Resolver) ResolvePath(fpath string) (*merkledag.Node, error) {
 		return nil, err
 	}
 
-	return s.ResolveLinks(nd, parts[1:])
+	return s.ResolveLinks(nd, parts)
 }
 
 // ResolveLinks iteratively resolves names by walking the link hierarchy.
 // Every node is fetched from the DAGService, resolving the next name.
-// Returns the last node found.
+// Returns the list of nodes forming the path, starting with ndd. This list is
+// guaranteed never to be empty.
 //
 // ResolveLinks(nd, []string{"foo", "bar", "baz"})
 // would retrieve "baz" in ("bar" in ("foo" in nd.Links).Links).Links
 func (s *Resolver) ResolveLinks(ndd *merkledag.Node, names []string) (
```

Comment:
We need this because when we PUT to /ipfs/HASH/dir/dir/file, the hash of file changes. This makes necessary to update the directories in succession up to the root hash. I felt it was better not to parse the path two times (once for getting the leaf file hash, one other time to rewrite the directories). Instead the path parsing is done once using the ResolveLinks function